### PR TITLE
Restructure project files

### PR DESCRIPTION
**Pull request by AquaGeek**
This pull request shuffles a lot of things around, with the end goals of:

1. Making tests first-class citizens
2. Supporting the Swift Package Manager (eventually)
3. Supporting Carthage (if folks want to use that; support hasn't been added yet)

Basically, I created an Xcode project for the library itself and moved the tests from the (now renamed) example project.

I also made Starscream a git submodule, because I think that provides the most flexibility for development without hindering package manager choices (Package.swift will just add it as a dependency, Carthage consumers will add it to their own Cartfile, etc). This definitely warrants further discussion to see if there is another way we want to go about doing this.

## TODO
- [X] Move all source classes to `Source/CLASS_NAME.swift` instead of `Pod/Classes/**`
- [X] Move all classes into individual files: `Socket.swift`, `Channel.swift`, etc.
- [X] Remove `Phoenix` namespacing struct
- [ ] Fix Travis
- [X] Remove _Pod.xcodeproj
- [ ] ~~Enable Carthage~~ (this will be done in a subsequent diff)
- [x] Update documentation

### CODE REVIEW

**Comment by davidstump** [comment]
This is a great start. I had started on a refactor to restructure the repo in a reasonably significant manner. Many of these changes are headed down that same path. Items I would like to discuss or tackle in addition to these changes in a restructure PR would be:

- [ ] Move all source classes to `Source/CLASS_NAME.swift` instead of `Pod/Classes/**`
- [ ] Move all classes into individual files: `Socket.swift`, `Channel.swift`, etc.
- [ ] Remove Phoenix struct. Holdover from initial attempt to "modularize" the repo, but doesnt really fall in line with most swift dev patterns I come across. 
- [x] Fix Travis
- [x] Remove _Pod.xcodeproj
- [ ] Enable Carthage (could be another PR if better suited)
- [ ] Update documentation (could also be another PR, but need better direction in docs as well as updated requirements/compatability). 

Thats my initial list - let me know if you think we should add/remove any of those items. I would love to hear any opinions on these or other restructuring ideas.

Cheers!

**Comment by AquaGeek** [comment]
> Move all source classes to Source/CLASS_NAME.swift instead of Pod/Classes/**

This is already done.

> Move all classes into individual files: Socket.swift, Channel.swift, etc.
> Remove Phoenix struct. Holdover from initial attempt to "modularize" the repo, but doesnt really fall in line with most swift dev patterns I come across.

Agreed on both. I'll update the PR.

> Enable Carthage (could be another PR if better suited)

I'd push to do this in a subsequent PR.

> Update documentation

Agreed.

Are you OK with the submodule approach for pulling in Starscream? This is only needed for development of the library itself — consumers would just use their package manager of choice to add it as a dependency as well.

**Comment by davidstump** [comment]
I am torn on the submodule approach.  In a perfect world, I like not having to worry about it when I start a new project. Add `pod "SwiftPhoenixClient"`, install, and start coding.

If Starscream becomes less than ideal (for some reason), I like the idea of being able to change the dependency and anyone that updates the package continues using SwiftPhoneixClient without worrying about the underlying websocket engine. 

Thoughts?

**Comment by AquaGeek** [comment]
The submodule is only needed if you're working on this library itself. If you're just building an app that leverages this library, you should only need to add `pod "SwiftPhoenixClient"` to your Podfile and `pod install`.

**Comment by davidstump** [comment]
Ah ok. I thought you were proposing that the user would need to add pod “Starscream” to their own Podfile in order to pull the dependency in. 

Having a submodule just for local dev seems fine by me.

File: Source/Models.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
+//
+//  Models.swift
```
Comment (davidstump) [inline Source/Models.swift]: Totally might be jumping the gun, but any specific reason to keep these classes grouped versus in their own respective swift files?

Comment (davidstump) [inline Source/Models.swift]: Sure, if you don't mind. Primarily just to make it easier to nail down what happens where if someone wants to debug something or contribute. Thanks :)

Comment (davidstump) [inline Source/Models.swift]: Thanks!! 👍

**Comment by davidstump** [comment]
Based on your checklist, the only item remaining is Travis (which we can push to another PR easily). Do you want me to pull this down and give it a spin yet?

Thanks a ton for all your work - these are some awesome and long-needed changes!

**Comment by AquaGeek** [comment]
This should be pretty much wrapped up. Feel free to pull and try it out.

**Pull request merged**